pkg/k8s/ingresses/translate.go

```diff
@@ -0,0 +1,132 @@
+package ingresses
+
+import (
+	"github.com/okteto/okteto/pkg/model"
+	networkingv1 "k8s.io/api/networking/v1"
+	networkingv1beta1 "k8s.io/api/networking/v1beta1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/intstr"
+)
+
+// Translate Endpoint to Ingress
+// Translate Service to Ingress
+
+type TranslateOptions struct {
+	Name      string
+	Namespace string
+}
+
+// TranslateEndpoint translates the endpoints spec at compose or okteto manifest and returns an ingress
+func TranslateEndpoint(endpointName string, endpoint model.Endpoint, opts *TranslateOptions) *Ingress {
```

Comment:
Minor comment: As this is translating an ingress, I would call it `Translate` (without the endpoints, it could be misleading), we are creating the ingress object with the specified endpoints

Reply:
I'm more into expressive names like `TranslateEndpointToIngress`, but it doesn't match the code style we have. Since this is inside `ingresses` package, I'd go with `TranslateFromEndpoint`